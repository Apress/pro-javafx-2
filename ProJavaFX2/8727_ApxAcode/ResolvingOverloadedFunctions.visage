/*
 * @test
 * @run
 */

function f(i:Integer, n:Number):Void {
  println("Integer, Number version of f is called with ({i}, {n}).");
}

function f(n:Number, i:Integer):Void {
  println("Number, Integer version of f is called with ({n}, {i}).");
}

f(1024, 3.14);
f(6.28, 2048);
// f(1024, 2048); // Won't compile
