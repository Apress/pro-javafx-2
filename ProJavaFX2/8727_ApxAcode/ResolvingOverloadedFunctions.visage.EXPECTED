Integer, Number version of f is called with (1024, 3.14).
Number, Integer version of f is called with (6.28, 2048).
